samcli/commands/init/__init__.py

```diff
@@ -64,21 +65,31 @@ def cli(ctx, location, runtime, output_dir, name, no_input):
         \b
         $ sam init --location /path/to/template/folder
 
+        \b
+        Initializes a new SAM project using custom template with additional known context
+        \b
+        # Add multiple -e for multiple context to be added (-e some=Value -e other=Value)
+        $ sam init --location gh:aws-samples/cookiecutter-aws-sam-python -e project_short_description=Oi
```

Comment:
This makes a lot of sense, for new cookiecutter samples, that have other content can be subbed.